src/main/java/org/folio/rest/impl/CheckinReceivePiecesHelper.java

```diff
@@ -737,4 +726,19 @@ private boolean isMissingLocation(PoLine poLine, Piece piece) {
       }
     });
   }
+
+  CompletionStage<List<CompletableFuture<Void>>> removeForbiddenEntities(List<PoLine> poLines, Map<String, Map<String, T>> pieces) {
+    Map<String, List<PoLine>> poLinesGroupedByOrderId = poLines.stream().collect(groupingBy(PoLine::getPurchaseOrderId));
+    String query = buildQuery(convertIdsToCqlQuery(new ArrayList<>(poLinesGroupedByOrderId.keySet())), logger);
+    String url = String.format(GET_PURCHASE_ORDERS, Integer.MAX_VALUE, 0, query, lang);
+    return handleGetRequest(url, httpClient, ctx, okapiHeaders, logger)
+      .thenApply(orders -> orders.mapTo(PurchaseOrders.class).getPurchaseOrders().stream().map(order -> protectionHelper.isOperationRestricted(order.getAcqUnitIds(), ProtectedOperationType.UPDATE)
```

Comment:
Consider to move processing logic to separate method or at least refactor it to be more readable e.g. move `orders -> orders.mapTo(PurchaseOrders.class).getPurchaseOrders()` as separate step.